pkg/client/unversioned/helper.go

```diff
@@ -408,7 +419,14 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 		return nil, err
 	}
 
-	client := NewRESTClient(baseURL, config.Version, config.Codec, config.QPS, config.Burst)
+	// baseURL is versioned, let's get a base without the version
```

Comment:
It seems like we should fix our internal structure to be more reasonable and keep the unreasonable `baseURL` contained to the edges where we accept input.